libs/core/src/components/sage-switch/sage-switch.tsx

```diff
@@ -0,0 +1,150 @@
+import { Component, Element, Event, EventEmitter, Host, h, Prop } from '@stencil/core';
+
+@Component({
+  tag: 'sage-switch',
+  styleUrl: 'sage-switch.scss',
+  shadow: true,
+})
+export class SageSwitch {
+  @Element() el: HTMLSageSwitchElement;
+
+  /**
+   * A unique identifier for this input. Associates the input with its label
+   */
+  @Prop() componentId!: string;
+
+  /**
+   * Sets state of input control
+   */
+  @Prop() checked = false;
+
+  /**
+   * Disables input control
+   */
+  @Prop() disabled? = false;
+
+  /**
+   * Text displayed with invalid state
+   */
+  @Prop() errorMessage?: string;
+
+  /**
+   * Text used for additional control description
+   */
+  @Prop() helperMessage: string;
+
+  /**
+   * Indicates when user input is not valid
+   */
+  @Prop() invalid? = false;
+
+  /**
+   * Text to be displayed as the switch label
+   */
+  @Prop() label!: string;
+
+  /**
+   * Used for identifying form data. Unifies a group of radio inputs for toggling a single property/value
+   */
+  @Prop() name: string;
+
+  /**
+   * Forces the input to be required
+   */
+  @Prop() required? = false;
+
+  /**
+   * Specifies the underlying input element type
+   * @defaultValue 'checkbox'
+   */
+  @Prop() type: 'checkbox' | 'radio' = 'checkbox';
+
+  /**
+   * Attribute sent in form data. Mainly used to distinguish radio inputs
+   */
+  @Prop() value: string;
+
+  /**
+   * Event emitted on input change
+  */
+  @Event() sageSwitchChange: EventEmitter<InputEvent>;
+
+  private onSwitchUpdate = (e: Event) => {
+    if (this.disabled) return;
+    this.sageSwitchChange.emit(e as InputEvent);
```

Comment:
Exposes full Event object ([see example in docs](https://github.com/Kajabi/sage/pull/55/files#diff-67621dfba256770b27076b079513ef884b1122961f1c6c57ee0ae730beee548dR21-R29))